docs/reports/marin-8b-retro.md

```diff
@@ -0,0 +1,688 @@
+- *Jellyfish (First Cooldown)*: It was time to cooldown as we were starting to run low on DCLM. Following recent work on midtraining (e.g. [Olmo 2](https://arxiv.org/abs/2501.00656)), we decided to fold in higher quality data during cooldown. (3.78T→4.78T tokens)
+- *Phoenix (Reheated)*: We had more time for training, so we rapidly rewarmed the model and transitioned our mixture to [Nemotron-CC](https://arxiv.org/abs/2412.02595) (plus [StarCoder Data](https://huggingface.co/datasets/bigcode/starcoderdata)). (4.78T→11.1T tokens)
+- *Starling (Second Cooldown)*: Now we were running low on time, so we started another cooldown. We followed a similar process to the first cooldown, but added a few new datasets that we had created and also some that had dropped since our previous attempt. (11.1T→12.75T tokens)
+
+These phases were not planned in advance. Decisions were made reactively based on changing timelines and data availability.
+
+While the final Marin 8B model came from a single linear run, we also explored a number of side paths. Some of these were by design—for example, short microannealing experiments
+(inspired by [Olmo 2](https://arxiv.org/abs/2501.00656), [Yi](https://arxiv.org/html/2403.04652v1), [DBRX](https://www.databricks.com/blog/introducing-dbrx-new-state-art-open-llm)) that informed our cooldown mix.
+Others tested "deeper cooldown" strategies or addressed issues like "SFT-ability." We summarize these later, but you can find more detail in:
+
+**Annealing Experiments:**
+* [GH#784](https://github.com/marin-community/marin/issues/784): High Quality Data Annealing Experiments on Dolma/Dolmino
+* [GH#820](https://github.com/marin-community/marin/issues/820): Evaluate High Quality Datasets for Cooldown Mix
+* [GH#934](https://github.com/marin-community/marin/issues/934): Comparing Cooldowns on Pretraining Data v.s. HQ Data
+
+**SFT-ability:**
+* [GH#898](https://github.com/marin-community/marin/issues/898): Raccoon: Try deepening the cooldown of "jellyfish" to see if it improves SFT
+* [GH#916](https://github.com/marin-community/marin/issues/916): Spoonbill: not-quite-so-deep cooldown
+
+
+## Phase 1: Kestrel (WSD-S Phase)
+
+In the first phase, we trained from scratch using what was, at the time, the best publicly available dataset:
+[DCLM Baseline](https://huggingface.co/datasets/mlfoundations/dclm-baseline-1.0), from the [DCLM paper](https://arxiv.org/abs/2406.11794).
+We adopted their “best mixture”: DCLM Baseline, [StarCoder Data](https://huggingface.co/datasets/bigcode/starcoderdata), and [Proofpile 2](https://huggingface.co/datasets/EleutherAI/proof-pile-2).
+
+The DCLM paper used a curriculum, mixing in StarCoder and Proofpile only near the end of training.
+We did **not** start with a curriculum—though we ended up adding quite a bit of one later.
```

Comment:
why didn't we do a curriculum again?  I think there was a desire to keep things simple?  We also didn't add all the stuff in Dolma...

Reply:
It might be worth just noting that at the time, we were trying to prepare our own datasets of everything (StackExchange, Reddit, arxiv, Wikipedia), so I think it was more like we wanted to replicate DCLM while we had the compute while the data processing was still happening.

Reply:
Oh I guess you sort of say this later